include/sys/sys_heap.h

```diff
@@ -61,6 +61,13 @@ struct z_heap_stress_result {
 	uint64_t accumulated_in_use_bytes;
 };
 
+struct sys_heap_stats {
+	uint32_t avail_chunks;
+	uint32_t avail_chunks_min;
+	uint32_t avail_contig;
+	uint32_t avail_contig_min;
```

Comment:
There is probably a big bikeshed waiting with this API.  Lots of people are going to want answers in bytes instead of chunks, and to have the code subtract the chunk header overhead from the answer.  Personally I prefer it this way for efficiency and clarity.